### ISSUE TITLE
Tensor.expand does not respect torch.no_grad()

### ISSUE THREAD

**Issue by fritzo** [pytorch/pytorch#25932]
## 🐛 Bug

`Tensor.expand` does not detach when called inside `torch.no_grad()`.

## To Reproduce

```py
x = torch.tensor(0., requires_grad=True)
with torch.no_grad():
    y = x.expand(1)
assert not y.requires_grad  # fails
```

## Expected behavior

Like all other torch operations, `.expand()` should respect `torch.no_grad()`.

## Environment

```
$ python ~/pytorch/torch/utils/collect_env.py
Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.5
GCC version: Could not collect
CMake version: version 3.12.0

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.17.1
[pip] torch==1.2.0
[pip] torchfile==0.1.0
[pip] torchvision==0.4.0
[conda] torch                     1.2.0                    pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchvision               0.4.0                    pypi_0    pypi
```

cc @ezyang @SsnL @albanD @zou3519

**Comment by fritzo** [comment]
cc @neerajprad @iffsid

**Comment by SsnL** [comment]
Duplicate of https://github.com/pytorch/pytorch/issues/11390

**Comment by fmassa** [comment]
Closing following @SsnL comment, let's keep this discussion in #11390 , and @albanD is assigned to work on it.
